src/wallet/wallet.cpp

```diff
@@ -41,7 +41,7 @@ CAmount maxTxFee = DEFAULT_TRANSACTION_MAXFEE;
 unsigned int nTxConfirmTarget = DEFAULT_TX_CONFIRM_TARGET;
 bool bSpendZeroConfChange = true;
 bool fSendFreeTransactions = false;
-bool fPayAtLeastCustomFee = true;
+bool fPayAtLeastCustomFee = false;
```

Comment:
Yes, this PR depends on and replaces #6649, thus fixes #6479.

I'd propose to rename this PR to "Default fPayAtLeastCustomFee to false" and go with Option 1? It seems better to get rid of this fixed-fee-regardless-of-tx-size approach completely. To cite @sipa :

> [fixed fee amount per transaction] wouldn't make sense. You're paying for space inside blocks, and miners prioritize transactions based on fee per byte. As you don't know the size of a transaction before creating it, specifying its exact fee would likely lead to unexpected behaviour.
> –  Pieter Wuille Aug 5 at 12:40 [[1]](https://bitcoin.stackexchange.com/questions/38957/paytxfee-vs-maxtxfee-which-one-overrides-the-other#comment45766_38958)